shell/components/InputOrDisplay.vue

```diff
@@ -18,31 +17,37 @@ export default {
       default: 'edit'
     }
   },
-  setup(props, { slots }) {
-    const isView = computed(() => props.mode === _VIEW);
+  computed: {
+    isView() {
+      return this.mode === _VIEW;
+    },
 
-    const displayValue = computed(() => {
-      if (Array.isArray(props.value) && props.value.length === 0) {
+    displayValue() {
+      if (Array.isArray(this.value) && this.value.length === 0) {
         return '';
       } else {
-        return props.value;
-      }
-    });
-
-    return () => {
-      if (isView.value) {
-        return h('div', { class: 'label' }, [
-          h('div', { class: 'text-label' }, slots.name ? slots.name : props.name),
-          h('div', { class: 'value' }, slots.value ? slots.value : displayValue.value)
-        ]);
-      } else {
-        return slots.default;
```

Comment:
Alternatively, we could have invoked the slot as a function via:

```
return slots.default();
```

The same would go for the two slots above.

Reply:
Thanks Phil, better to know what the error was, I found another case where we need to update the slots API. https://github.com/rancher/dashboard/pull/12005